include/pmacc/Environment.def

```diff
@@ -32,5 +33,100 @@ namespace pmacc
     {
         struct Environment;
 
+        /** collect state variables of the environment context
+         *
+         * This class handle the initialization and finalize of the
+         * MPI context and the selection of the GPU.
+         */
+        class EnvironmentContext
```

Comment:
This class is moved into the `def` file because it is required to take care that the transaction system is not started before we selected a device.
Moving this class here is no problem because it is not pulling files with include side effects.